spinalcordtoolbox/scripts/sct_image.py

```diff
@@ -324,6 +330,41 @@ def main(argv=None):
         dim = dim_list.index(dim)
         im_out = split_data(im_in, dim)
 
+    elif arguments.stitch is not None:
+        # preserve original orientation
+        orig_ornt = im_in_list[0].orientation
+        # reorient images to PIL to use stitching module
+        im_ornt = []
+
+        # use SCT to reorient images
+        for idx, i in enumerate(arguments.i):
+            # TODO: replace 0.nii.gz, etc. by temp files
+            # TODO: where does SCT save this, temp dir?
```

Comment:
Hi @joshuacwnewton,

I tried to implement this structure, but I believe it currently does not work as intended:

`path_tmp = tmp_create(basename="image-stitching")`

Running `sct_image -i img1.nii.gz -img2.nii.gz -o stitched.nii.gz -stitch` throws an error, if I try to pass the temp folder to the `stitching` command line tool.

`** ERROR (nifti_image_write_hdr_img2): cannot open output file ' /tmp/sct-20220824152736.139553-image-stitching-92e5i614/stitched.nii.gz'`

I have the feeling this is caused by the temp directory being only visible to the current process, but if we spawn a new shell (in order to call `stitching`), we can no longer access the temporary folder as this shell does not have the privileges to enter/search/change the directory as it's using `mkdtemp`?

c.f. `User-callable function to create and return a unique temporary directory. [...]
The directory is readable, writable, and searchable only by the creating user.`

Either that, or there is an error in my implementation :slightly_smiling_face:

Reply:
Please let me know what you think and feel free to review and provide feedback for the updated implementation!

Reply:
> `** ERROR (nifti_image_write_hdr_img2): cannot open output file ' /tmp/sct-20220824152736.139553-image-stitching-92e5i614/stitched.nii.gz'`

Hmm... I notice that the error message contains an extra space at the start of the path: ` /tmp/`.

I also notice that in the call to stitching, right now it says `'-i '` instead of `'-i'`.

I'm wondering if that extra space might be the cause? 🤔

Reply:
> I also notice that in the call to stitching, right now it says `'-i '` instead of `'-i'`.

good catch! Unfortunately this does not fix it completely :slightly_frowning_face: 

I believe the function call to the stitching module is correct, at least the terminal print out says (or am I misinterpreting it?):

`/home/juli/git_repositories/sct_development/spinalcordtoolbox/bin/stitching -i 'img1.nii.gz img2.nii.gz' '-o /tmp/sct-20220824161453.799757-image-stitching-goawicp5/stitched.nii.gz' -a # in /tmp/sct-20220824161453.799757-image-stitching-goawicp5
`

Reply:
Ah, there's another error with the output argument: the `-o` and the following filename `/tmp/sct-...` are enclosed in the same set of single quotes, which will definitely cause problems.

Reply:
@mguaypaq perfect - it's running now end to end! thank you very much for discovering this. Fix c.f. [df9521b](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3865/commits/df9521b5a75336fbfefb4010d2dc3683fd5b62b0)